service/forward_service.cc

```diff
@@ -96,20 +98,26 @@ static shared_ptr<cql3::selection::selection> mock_selection(const std::vector<q
     return cql3::selection::selection::from_selectors(db, schema, std::move(raw_selectors));
 }
 
-future<query::forward_result> forward_service::execute(query::forward_request req) {
+future<query::forward_result> forward_service::execute(query::forward_request req, std::optional<tracing::trace_info> tr_info) {
+    tracing::trace_state_ptr tr_state;
+    if (tr_info) {
+        tr_state = tracing::tracing::get_local_tracing_instance().create_session(*tr_info);
+        tracing::begin(tr_state);
+    }
+
+    tracing::trace(tr_state, "Executing forward_request");
+
```

Comment:
Please fold into relevant patches.

Reply:
Do you mean that I have to fold 153d3a8fd17185b5dca8ffa2b01dbc66f057b4fc into 04d4554cbc7e2d4fb82cf7dd94730cca6bd64c85? Why? I thought splitting commits into the small atomic changes is ok.

Reply:
Splitting unrelated changes into atomic commits is highly encouraged. Here, the changes are not unrelated, tracing is an integral part of query execution.

I'm okay with keep it separated though.